core/src/main/scripts/import_data_validator/validateData.py

```diff
@@ -1422,13 +1427,139 @@ class SegValidator(Validator):
+        parsed_coords = {}
+        for col_index, col_name in enumerate(self.cols):
+            value = data[col_index]
+            if col_name == 'ID':
                 self.checkSampleId(value, column_number=col_index + 1)
+            elif col_name == 'chrom':
+                if value in self.chromosome_lengths:
+                    parsed_coords[col_name] = value
+                else:
+                    self.logger.error(
+                        ('Unknown chromosome, must be one of (%s)' %
+                         '|'.join(self.chromosome_lengths.keys())),
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            elif col_name in ('loc.start', 'loc.end'):
+                try:
+                    parsed_coords[col_name] = int(value)
+                except ValueError:
+                    self.logger.error(
+                        'Genomic position is not an integer',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+                    # skip further validation specific to this column
+                    continue
+                # 0 is the first base, and loc.end is not part of the segment
+                # 'chrom' has already been read, as column order is fixed
+                if parsed_coords[col_name] < 0 or (
+                        'chrom' in parsed_coords and
+                        parsed_coords[col_name] > self.chromosome_lengths[
+                                                      parsed_coords['chrom']]):
+                    self.logger.error(
+                        'Genomic position beyond end of chromosome '
+                        '(chr%(chr)s:0-%(end)s)',
+                        {'chr': parsed_coords['chrom'],
+                         'end': self.chromosome_lengths[
+                                    parsed_coords['chrom']]},
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+                    # not a valid coordinate usable in further validations
+                    del parsed_coords[col_name]
+            elif col_name == 'num.mark':
+                if not self.checkInt(value):
+                    self.logger.error(
+                        'Number of probes is not an integer',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            elif col_name == 'seg.mean':
+                if not self.checkFloat(value):
+                    self.logger.error(
+                        'Mean segment copy number is not a number',
```

Comment:
check float and checkint is the type of low level validation that I would like to leave over to the DB. This validation script is supposed to check higher level mistakes in the set of files, mistakes that are hard to catch with a DB design (or hard to translate into a clear error message to the user). I think that adding these checks here is over doing it in the wrong place. 

Also, if this would be a requirement, we should do it in a generic way where the types of the fields are stored in the same dictionary that now holds whether they are optional or not. And then the validation could be done in the super class for all types. But again, let's *not* do this.